web-registry/src/features/marketplace/BuySellOrderFlow/BuySellOrderFlow.tsx

```diff
@@ -107,23 +97,95 @@ export const BuySellOrderFlow = ({
     error,
     setError,
   } = useMsgClient(handleTxQueued, handleTxDelivered, handleError);
+
   const accountAddress = wallet?.address;
   const txHash = deliverTxResponse?.transactionHash;
   const txHashUrl = getHashUrl(txHash);
   const errorEnum = findErrorByCodeEnum({ errorCode: error });
   const ErrorIcon = errorsMapping[errorEnum].icon;
 
-  const project = useMemo(
-    () => ({
-      id: selectedProject?.id.toString() ?? '',
-      sellOrders: projectUiSellOrdersInfo?.filter(
+  /**
+   * data processing (business logic)
+   */
+  const [selectedProject, setSelectedProject] =
+    useState<ProjectWithOrderData>();
+
+  useEffect(() => {
+    if (projects?.length === 1) setSelectedProject(projects[0]);
+  }, [projects]);
+
+  const setSelectedProjectById = (projectId: string): void | undefined => {
+    if (!projects || projects.length <= 1) return;
+    const found = projects.find(project => project.id === projectId);
+    if (found) setSelectedProject(found);
+  };
+
+  const sellOrderIds = useMemo(
+    () =>
+      projects &&
+      projects
+        .map(project => project.sellOrders)
+        .flat()
+        .map(sellOrder => sellOrder.id),
+    [projects],
+  );
+
+  const { sellOrdersResponse, refetchSellOrders } = useQuerySellOrders();
+  const allSellOrders = sellOrdersResponse?.sellOrders;
+
+  const projectUiSellOrdersInfo = useMemo(
```

Comment:
```suggestion
  const uiSellOrdersInfo = useMemo(
```

Not sure we need project in the variable name here.